Clarify description of `safe` option

shannonxtreme opened the pull request:
<!--
  Thanks for creating a Pull Request! Before you submit, please make sure
  you've done the following:

  - I read the contributing document at https://jekyllrb.com/docs/contributing/
-->

<!--
  Make our lives easier! Choose one of the following by uncommenting it:
-->

<!-- This is a 🐛 bug fix. -->
<!-- This is a 🙋 feature or enhancement. -->
This is a 🔦 documentation change.
<!-- This is a 🔨 code refactoring. -->

<!--
  Before you submit this pull request, make sure to have a look at the following
  checklist. If you don't know how to do some of these, that's fine! Submit
  your pull request and we will help you out on the way.

  - I've added tests (if it's a bug, feature or enhancement)
  - I've adjusted the documentation (if it's a feature or enhancement)
  - The test suite passes locally (run `script/cibuild` to verify this)
-->

## Summary

Change `Safe` option description to improve clarity and specificity. 

## Context

Resolves #8321

ashmaroli commented:
While we're at it, I think it'd be better if we also clarify that it is not just *custom plugins* that are disabled, but also plugin-gems that have not been listed under the `whitelist` key.

shannonxtreme commented:
Does this mean all plugins not in the whitelist, regardless of whether "custom" or a gem, are disabled for safe mode?

ashmaroli commented:
> Does this mean all plugins not in the whitelist, regardless of whether "custom" or a gem, are disabled for safe mode?

Technically, yes.
But there is a catch, though. This screening is only applicable for plugins loaded via the config file.
If one lists a gem under the `:jekyll-plugins` group in the Gemfile, it will be loaded irrespective of the `safe` option.
The above observation is just for your information. It need not be included in this PR to avoid unnecessary confusion.

shannonxtreme commented:
@ashmaroli I added a clarification and removed "custom" from plugins to make it less confusing. The gist of it is that unless whitelisted, plugins are disabled. Didn't add anything about gems in `:jekyll-plugins` but you can modify the line to

 "Whitelisted plugins as well as gems listed in `:jekyll-plugins` on the Gemfile are not disabled." 

if you want.

ashmaroli commented on docs/_docs/configuration/options.md:
```diff
@@ -44,8 +44,9 @@ class="flag">flags</code> (specified on the command-line) that control them.
         <p class="description">
-          Disable <a href="/docs/plugins/">custom plugins</a>, caching to disk
-          and ignore symbolic links.
+          Disable <a href="/docs/plugins/">plugins</a>, caching to disk
+          and ignore symbolic links.  </br>
+          Whitelisted plugins are not disabled.
```
How about rephrasing to the following?
```suggestion
          Disable <a href="/docs/plugins/">non-whitelisted plugins</a>, caching to disk
          and ignore symbolic links.
```

shannonxtreme replied:
I like it. The only reason I had it separate was to link "whitelisted" to the config page for whitelisting.

ashmaroli replied:
I don't think we have a dedicated section on the `whitelist` configuration..
If that's indeed the case, I think you can add a *note* to the page at `/docs/plugins/` informing the visitor about the whitelist..
(*it can be part of this PR itself*)

ashmaroli commented:
Thank you @shannonxtreme 
@jekyllbot: merge +doc
